File: dinamic/Largest_overall_subsequence.cpp
#include <fstream>
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
#define MAX(a,b) ((a) > (b) ? (a) : (b))

using namespace std;
int main()
{
    ifstream fin("input.txt");
    ofstream fout("output.txt");
    string A;
    string B;
    fin >> A;
    fin >> B;
    int n = A.length();
    int m = B.length();
    int i = 1;
    int j = 1;
    int F[1000+1][1000 + 1] = {};
    for (i; i < n + 1; i++)
    {
        for (j; j < m +1; j++)
        {
            if (A[i -1] == B[j - 1])
            {
                F[i][j] = F[i - 1][j - 1] + 1;
            }
            else 
            {
               F[i][j] = MAX(F[i - 1][j], F[i][j - 1]);
            }
        }
    }
    i = n;
    j = m;
    vector<char> sequence;
    //вывод найденной последовательности
    while (i*j > 0)
    {
        if (A[i - 1] == B[j -1])
        {
            sequence.push_back(A[i - 1]);
            i--;
            j--;
            
        }
        else if (F[i - 1][j] == F[i][j])
        {
            /* code */i--;
        }
        
        
    }
    cout << sequence[0];
    cout << sequence[1];
    cout << sequence[2];
    cout << sequence[3];
}

File: dinamic/Largest_overall_subsequence.py
fin  = open('input.txt', 'r').read().split('\n')         
fout = open('output.txt', 'w')
A, B = fin[0],fin[1]
n = len(A)
m = len(B)
sub_count = [[0] * (m + 1) for i in range(n + 1)]
for i in range(1, n + 1):
    for j in range(1, m + 1):
        if A[i - 1] == B[j - 1]:
            sub_count[i][j] = sub_count[i - 1][j - 1] + 1 
        else: 
            sub_count[i][j] = max(sub_count[i - 1][j], sub_count[i][j - 1]) 
sub = []
i = n
j = m
while i > 0 and j > 0:
    if A[i - 1] == B[j - 1]:
        sub.append(A[i - 1])
        i -= 1
        j -= 1
    elif sub_count[i - 1][j] == sub_count[i][j]:
        i -= 1 
    else: 
        j -= 1 
sub = sub[::-1]
fout.write(''.join(sub) )